App.tsx

```diff
@@ -0,0 +1,44 @@
+import './polyfills';
+import React from 'react';
+import { StatusBar } from 'expo-status-bar';
+import { WagmiConfig } from 'wagmi';
+import { mainnet, polygon, arbitrum, polygonMumbai } from 'wagmi/chains';
+import {
+  createWeb3Modal,
+  defaultWagmiConfig,
+  Web3Modal,
+} from '@web3modal/wagmi-react-native';
+import HomePage from './src/pages/HomePage';
+
+const projectId = process.env.EXPO_PUBLIC_WALLETCONNECT_CLOUD_PROJECT_ID;
+
+const metadata = {
+  name: 'Example dApp',
+  description: 'Modern Example dApp from Callstack',
+  url: 'https://callstack.com/',
+  icons: ['https://avatars.githubusercontent.com/u/42239399'],
+  redirect: {
+    native: 'YOUR_APP_SCHEME://',
+    universal: 'YOUR_APP_UNIVERSAL_LINK.com',
+  },
+};
+
+const chains = [mainnet, polygon, polygonMumbai, arbitrum];
+
+const wagmiConfig = defaultWagmiConfig({ chains, projectId, metadata });
+
+createWeb3Modal({
```

Comment:
Let's add some code comment what is the next line actually doing.

Reply:
BTW why is it at top level scope?

Reply:
Added comments. @mdjastrzebski What do you mean with the top level scope, would you rather put in another file?

Reply:
Just wondering, however that seems like a pattern from WC docs: https://docs.walletconnect.com/web3modal/react-native/about#implementation

Just found it a bit weird that it's a top level `createXxx` call and we do not capture any retuned value. I would add comment `// Initialized WalletConnect Web3Modal library.`

Reply:
Yeah 👌🏻 comment is already there!